modules/microshift-greenboot-create-health-check-script.adoc

```diff
@@ -1,57 +1,114 @@
+* You can place shell scripts you want to run after a boot has been declared failed in `/etc/greenboot/red.d`. For example, if you have steps to heal the system before restarting, you can create scripts for your use case and place them in the `/etc/greenboot/red.d` directory.
 
-. To create a health check script, run the following command:
-+
-[source, terminal]
-----
-$ SCRIPT_FILE=/etc/greenboot/check/required.d/50_busybox_running_check.sh
-sudo curl -s https://raw.githubusercontent.com/openshift/microshift/3b7f6025cd77bd1bf827416fd026783ead82b7c8/docs/config/busybox_running_check.sh \
-  -o ${SCRIPT_FILE} && echo SUCCESS || echo ERROR
-sudo chmod 755 ${SCRIPT_FILE}
-----
-+
-In this example, the script verifies that busybox is running as expected. You can replace `/etc/greenboot/check/required.d/50_busybox_running_check.sh` with your own workload details.
-+
-[NOTE]
+[id="microshift-greenboot-about-workload-health-check-script-example_{context}"]
+== About the workload health check script example
+
+The following example uses the {product-title} health check script as a template. You can use this example with the provided libraries as a guide for creating basic health check scripts for your applications.
+
+[id="microshift-greenboot-app-health-check-basic-prereqs_{context}"]
+=== Basic prerequisites for creating a health check script
+
+* The workload must be installed.
+* You must have root access.
+
+[id="microshift-greenboot-app-health-check-ex-reqs_{context}"]
+=== Example and functional requirements
+
+You can start with the following example health check script. Modify it for your use case. In your workload health check script, you must complete the following minimum steps:
+
+* Set the environment variables.
+* Define the user workload namespaces.
+* List the expected pod count.
+
+[IMPORTANT]
 ====
-In this example, the {product-title} core service health checks run before the user workload health checks.
+Choose a name prefix for your application that ensures it runs after the `40_microshift_running_check.sh` script, which implements the {product-title} health check procedure for its core services. Greenboot automatically names the script based on the application name.
 ====
 
-. To test that your script is running as expected:
-
-.. Restart the system.
+.Example workload health check script
 
-.. Once the system has restarted, run the following command:
-+
-[source, terminal]
-----
-$ sudo journalctl -o cat -u greenboot-healthcheck.service
+[source, bash]
 ----
-+
-.Example output for the busybox health check script
-+
-[source, terminal]
-----
-...
-...
-STARTED
-Waiting 300s for pod image(s) from the 'busybox' namespace to be downloaded
-Waiting 300s for 1 pod(s) from the 'busybox' namespace to be in 'Ready' state
-Checking pod restart count in the 'busybox' namespace
-FINISHED
-Script '50_busybox_running_check.sh' SUCCESS
+# #!/bin/bash
+set -e
+
+SCRIPT_NAME=$(basename $0)
+PODS_NS_LIST=(<namespace>)
+PODS_CT_LIST=(1      )
```

Comment:
We need to add a note that users must update these two lines with the info specific to their workloads. There can be one or more namespaces and their pod counts.
```suggestion
PODS_NS_LIST=(<user_workload_namespace1> <user_workload_namespace2>)
PODS_CT_LIST=(<user_workload_namespace1_pod_count> <user_workload_namespace2_pod_count>)
```